clients/src/main/java/org/apache/kafka/common/security/authenticator/LoginManager.java

```diff
@@ -55,13 +59,17 @@ public class LoginManager {
     private int refCount;
 
     private LoginManager(JaasContext jaasContext, String saslMechanism, Map<String, ?> configs,
-                         LoginMetadata<?> loginMetadata) throws LoginException {
+                 LoginMetadata<?> loginMetadata) throws LoginException {
         this.loginMetadata = loginMetadata;
         this.login = Utils.newInstance(loginMetadata.loginClass);
         loginCallbackHandler = Utils.newInstance(loginMetadata.loginCallbackClass);
         loginCallbackHandler.configure(configs, saslMechanism, jaasContext.configurationEntries());
         login.configure(configs, jaasContext.name(), jaasContext.configuration(), loginCallbackHandler);
-        login.login();
+        try (AutoCloseable loginResources = this::closeResources) {
```

Comment:
Should we move `login.configure` and `loginCallbackHandler.configure` into the try-catch block so that we cleanup if either fails?

Reply:
I was wondering about that, i think that's a sounded suggestion.